ingestion/src/metadata/ingestion/source/dashboard/tableau.py

```diff
@@ -250,7 +240,7 @@ def yield_dashboard_lineage_details(
         data_source = next(
             (
                 data_source
-                for data_source in self.workboook_datasources
+                for data_source in self.workbook_datasources
```

Comment:
You are running
```
self.workbook_datasources = json.loads(graphql_query_result.text)[
                    "data"
                ].get("workbooks")
```

But here you want to iterate on something that is potentially `None`.

Either handle `Option` and throw a logger or something, or use

```python
for data_source in self.workbook_datasources or []
```

to not fail on the iteration